src/main.rs

```diff
@@ -105,57 +107,78 @@ async fn webhook(
         return Ok(HttpResponse::NotFound().body("Not Found"));
     }
 
-    match core::spawn_job(&ctx.logs_dir, project.unwrap()) {
+    match core::spawn_job(project.unwrap(), ctx.clone()) {
         Ok(job_id) => Ok(HttpResponse::Ok().json(WebhookResponse { name: job_id })),
         Err(err) => Ok(err.error_response()),
     }
 }
 
 #[get("/jobs")]
 async fn get_jobs(ctx: web::Data<Context>) -> Result<HttpResponse> {
-    let log_dir = shellexpand::tilde(&ctx.logs_dir).into_owned();
-    let mut logs: Vec<String> = Vec::new();
-    let mut dir = async_std::fs::read_dir(log_dir).await?;
-
-    while let Some(entry) = dir.next().await {
-        let path = entry?.path();
-
-        let file_name = match path.file_name() {
-            Some(name) => name
-                .to_owned()
-                .into_string()
-                .map_err(|_err| SubiloError::ReadFileName {})?,
-            None => {
-                error!("Failed to read file at path {:?}", path);
-                continue;
-            }
-        };
+    let query = database::Query {
+        query: job::GET_ALL_JOBS_QUERY.to_owned(),
+        params: vec![],
+        map_result: |row| {
+            Ok(job::Job {
+                id: row.get(0)?,
+                name: row.get(1)?,
+                status: row.get(2)?,
+                started_at: row.get(3)?,
+                ended_at: row.get(4)?,
+            })
+        },
+    };
 
-        if file_name.ends_with(".json") {
-            logs.push(file_name.replace(".json", ""));
-        }
-    }
+    let jobs = ctx
+        .database
+        .send(query)
+        .await
+        .map_err(|err| SubiloError::DatabaseActor { source: err })?
+        .map_err(|err| SubiloError::Database { source: err })?;
 
-    Ok(HttpResponse::Ok().json(logs))
+    let res = HttpResponse::Ok().json(jobs);
+    Ok(res)
```

Comment:
[PP] :stuck_out_tongue_closed_eyes: 

```suggestion
    Ok(HttpResponse::Ok().json(jobs))
```

Reply:
😭